src/devices/timer.c

```diff
@@ -92,8 +92,8 @@ timer_sleep (int64_t ticks)
   int64_t start = timer_ticks ();
 
   ASSERT (intr_get_level () == INTR_ON);
-  while (timer_elapsed (start) < ticks) 
-    thread_yield ();
+  if (timer_elapsed (start) < ticks)
+    thread_sleep (ticks + start);
```

Comment:
`thread_yield()` docs : https://jhu-cs318.github.io/pintos-doxygen/html/thread_8c.html#a99b55ae8059d7834cc113334a3fd0c3c
> Yields the CPU.
The current thread is not put to sleep and may be scheduled again immediately at the scheduler's whim. 

`thread_sleep(int64_t ticks)`
> make thread status to `THREAD_BLOCKED` and add sleep queue